webui/react/src/components/UPlotChart.tsx

```diff
@@ -12,36 +14,41 @@ export interface Options extends Omit<uPlot.Options, 'width'> {
 interface Props {
   data?: AlignedData;
   options?: Options;
+  ref?: React.Ref<uPlot|undefined>;
 }
 
 const SCROLL_THROTTLE_TIME = 500;
 
-const UPlotChart: React.FC<Props> = ({ data, options }: Props) => {
+const UPlotChart: React.FC<Props> = forwardRef((
+  { data, options }: Props,
+  ref?: React.Ref<uPlot|undefined>,
+) => {
   const [ chart, setChart ] = useState<uPlot>();
   const chartDivRef = useRef<HTMLDivElement>(null);
+  const hasData: boolean = useMemo(() => !!(data && data[0].length > 0), [ data ]);
```

Comment:
**suggestion:** I just ran into a problem with `LearningCurveChart` where `uPlot` reports a problem if the series in the `data` don't match up in length with the `xValues`. So something like this will report an error:

```
[
  [ 1, 2, 3 ],
  [],
  [],
]
```

this will result in:
<img width="437" alt="Screen Shot 2021-04-27 at 12 07 28 PM" src="https://user-images.githubusercontent.com/220971/116309626-39040680-a766-11eb-8d73-fe76032c3630.png">
<img width="324" alt="Screen Shot 2021-04-27 at 12 07 39 PM" src="https://user-images.githubusercontent.com/220971/116309631-399c9d00-a766-11eb-86c0-6dfef1f25d47.png">

The reason this can happen is because sometimes series data can come in a bit later than xValues. Can we update this `hasData` to check that every row of data (xValues, and series data) all have the same length and not empty to be considered true?

Reply:
Do you have instructions to replicate the issue quickly? Just to be sure i properly address the fix when working on this.

Reply:
[test on this](http://localhost:3000/det/experiments/1289/visualization) once `LearningCurveChart` migrates over to `UPlotChart`